accepted/00000-upgrading-bootstrap.md

```diff
@@ -0,0 +1,70 @@
+
+The core proposition of this RFC can be summarized in two steps:  
+
+ 1. Simplify our existing stylesheets to reduce the use of custom overrides, and try and compose views and layouts using standard Bootstrap solutions. This step is beneficial even without the Bootstrap upgrade in mind, as currently our stylesheets are simply very convoluted and hard to reason about.  
+ 2. Upgrade Bootstrap. This includes many moving parts which will be outlined below.  
+
+Coming out successfully from the other side, we would hope to have simpler stylesheets with less to maintain, and fewer security issues.  
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Theme & styles
+
+Bootstrap 3 uses Less while Bootstrap 5 only offers Sass styles. This switch by itself is not very problematic, however there are additional issues that make the upgrade more difficult. Our current styles make heave use of numerous mixins from Bootstrap 3 which no longer exist in Bootstrap 5. Additionally, there are numerous breaking styling changes going from 3 to 4 and even more so moving from 4 to 5.  
+
+Instead of trying to manually migrate our existing very large theme source, I propose we instead try to first reduce our existing theme footprint as much as possible. We already have component scoped styles working and in use, so we can try move component or view specific styles (e.g. table styles, setup wizard etc) into component scoped styles and remove them from the global stylesheet.  
+
+This allows us to cut up the effort required and do at least some of the work gradually. I suspect the last core theme change will still be painful, though. It might be easier to rebuild the existing theme from the base Bootstrap style rather than migrating the existing one, but this is something we can consider after we've simplified the theme.  
+
+## Component and class name changes
+
+This is tied to the previous section, but worth discussing on its own. Numerous component, layout and utility classes have been dropped or changed in both upgrades, e.g. `input-group-addon` was changed in Bootstrap 4 and dropped in 5, so we need to use `input-group-text` and check if it renders correctly etc. The list is pretty long. As discussed in [this ticket](https://github.com/SUSE/spacewalk/issues/18346), it's unlikely that we'd find a tool that can handle all different ways we use Bootstrap (in JSP, JPSF, React, etc), however writing a migration script similar to the Typescript migration might be feasible. For this, we can assemble a list that covers all the breaking changes that affect us by going through the list of changes for 4 and 5, checking whether we use the affected classes, and if so, write down what the upgrade path is.  
```

Comment:
Mostly I would hope the same, that we can grep and replace after verifying we have the right set of matches. My main worry is dynamically created class names etc which I know we have in some places, but hopefully not much. Translations are a good point I did not consider, thanks for the heads up, I will keep it in mind.

Reply:
IIUC the most of the time we dynamically create class names is more often for `font-awesome` icons, but it might also be the case here and there for bootstrap ones. I don't expect that many but I'm afraid that yes, those will have to be caught and tackled manually.